h2o-algos/src/main/java/hex/tree/FriedmanPopescusH.java

```diff
@@ -216,9 +231,12 @@ static Frame computeFValues(int[] modelIds, Frame filteredFrame, String[] cols,
         Frame uniqueWithCounts = uniqueRowsWithCounts(filteredFrame);
         Frame uncenteredFvalues = new Frame(partialDependence(modelIds, uniqueWithCounts, learnRate, sharedTreeSubgraphs).vec(0));
         VecUtils.DotProduct multiply = new VecUtils.DotProduct().doAll(uniqueWithCounts.vec("nrow"), uncenteredFvalues.vec(0));
-        double meanUncenteredFValue = multiply.result / filteredFrame.numRows();
-        for (int i = 0; i < uncenteredFvalues.numRows(); i++) {
-            uncenteredFvalues.vec(0).set(i, uncenteredFvalues.vec(0).at(i) - meanUncenteredFValue);
```

Comment:
Every time we call `Vec.set` we call a lot of things that are not necessary in this usecase (e.g. recalculating stats). Using a `Vec.Writer` we can postpone these things until we are done with writing.